addon/components/new-bourbon-select-field.js

```diff
@@ -0,0 +1,214 @@
+import Component from '@ember/component';
+import EmberObject from '@ember/object';
+import { computed } from '@ember/object';
+import { isPresent } from '@ember/utils';
+import { A } from '@ember/array';
+import layout from '../templates/components/new-bourbon-select-field';
+
+export default Component.extend({
+  layout,
+  classNames: ['BourbonSelectField', 'NewBourbonSelectField'],
+  classNameBindings: [
+    'hasValue',
+    'fullWidth:btw-block',
+    'showList:BourbonSelectField--active',
+    'disabled:BourbonSelectField--disabled'
+  ],
+  attributeBindings: ['disabled:disabled'],
+
+  // passed in
+  content: null,
+  prompt: null,
+  value: null,
+  optionLabelPath: null,
+  optionValuePath: null,
+  optionEnabledPath: null,
+  action: null,
+
+  // internal
+  showList: false,
+  activeDescendant: null,
+  hasValue: computed.notEmpty('value'),
+
+  didRender() {
+    this.set(
+      'activeDescendant',
+      $('.NewBourbonSelectField-option[aria-selected]').attr('id')
+    );
+  },
+
+  ariaExpanded: computed('showList', function() {
+    return this.get('showList');
+  }),
+  _labelPath: computed('optionLabelPath', function() {
+    if (isPresent(this.get('optionLabelPath'))) {
+      return this.get('optionLabelPath').replace(/^content\.?/, '');
+    }
+  }),
+
+  _valuePath: computed('optionValuePath', function() {
+    if (isPresent(this.get('optionValuePath'))) {
+      return this.get('optionValuePath').replace(/^content\.?/, '');
+    }
+  }),
+
+  _enabledPath: computed('optionEnabledPath', function() {
+    if (isPresent(this.get('optionEnabledPath'))) {
+      return this.get('optionEnabledPath').replace(/^content\.?/, '');
+    }
+  }),
+
+  internalContent: computed(
+    'content',
+    '_labelPath',
+    '_valuePath',
+    '_enabledPath',
+    function() {
+      return A(
+        this.get('content').map(item => {
+          if (typeof item === 'string' || typeof item === 'number') {
+            return EmberObject.create({
+              label: item.toString(),
+              value: item,
+              enabled: true
+            });
+          } else {
+            let emberItem =
+              typeof item.get == 'function' ? item : EmberObject.create(item);
+
+            let endItem = {};
+            endItem.label = this.get('_labelPath')
+              ? emberItem.get(this.get('_labelPath'))
+              : emberItem.get('label');
+            endItem.value = this.get('_valuePath')
+              ? emberItem.get(this.get('_valuePath'))
+              : emberItem.get('value');
+            endItem.enabled = this.get('_enabledPath')
+              ? emberItem.get(this.get('_enabledPath'))
+              : true;
+
+            // when the value is supposed to be the item
+            if (endItem.value === undefined) endItem.value = emberItem;
+
+            return EmberObject.create(endItem);
+          }
+        })
+      );
+    }
+  ),
```

Comment:
Do we want to handle grouped content at this point? I'm imagining it being handled inside `internalContent`, and basically just concatenating the groups together, with headers set to `disabled=true`. I think that's how the current bourbon select field handles it?

Reply:
Right now we are only using grouped content for the search select.  The current `bourbon-select-field` doesn't support it.   I thought we could add grouped content when we we actually need it?